src/core/pal/feature.cpp

```diff
@@ -522,6 +523,12 @@ void createCandidateAtOrderedPositionOverPoint( double &labelX, double &labelY,
       break;
   }
 
+  // Take care of the label angle when creating candidates. See pr comments #44944 for details
+  // https://github.com/qgis/QGIS/pull/44944#issuecomment-914670088
+  QTransform transformRotation;
+  transformRotation.rotate( angle * QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::AngleRadians, QgsUnitTypes::AngleDegrees ) );
```

Comment:
```suggestion
  transformRotation.rotate( angle * 180 / M_PI );
```

Reply:
Because of consistency with the rest of the class or because of the little optimization?

Reply:
It's just overkill to use the qgis class for a fixed conversion like this